### PR TITLE
Makefile machinery work

### PR DESCRIPTION
**Pull request by ethteck**
`make setup` now fetches ido7 from our releases. Also, you can successfully run `make setup` if you already have a valid baserom.

### CODE REVIEW

**Comment by Roman971** [comment]
Can you explain how the ido 7.1 change would behave if you already have it locally?
And what happens if you can't download it for internet reasons or something?

**Comment by ethteck** [comment]
Sure thing. So, it tries to download ido 7.1 as part of `make setup` and will avoid replacing any existing files upon extraction. In other words, it won't change anything if you're already set up. If you can't download the file and are trying to run `make setup`, there may be an error..I didn't test that. Do you prefer a different approach? 

We could this step if the ido 7 dir exists. However, if there are files missing in the directory, this will prevent that from being fixed.

**Comment by Roman971** [comment]
In my opinion the make setup script should definitely run fine as long as you have the compiler, even if you can't download it. I also wouldn't want it to be redownloaded every time this command runs.

File: fixbaserom.py [hunk 1]
```diff
@@ -7,10 +7,10 @@
...
 # Determine if we have a ROM file
 romFileName = ""
-if (path.exists("baserom_original.z64")):
-    romFileName = "baserom_original.z64"
-elif (path.exists("baserom_original.n64")):
-    romFileName = "baserom_original.n64"
+if (path.exists("baserom.z64")):
+    romFileName = "baserom.z64"
+elif (path.exists("baserom.n64")):
+    romFileName = "baserom.n64"
```
Comment (Roman971) [inline fixbaserom.py:13]: Won't this change make it so it takes the existing baserom.z64 and override it? I don't understand why we would want to do that.
If we ask users to provide a baserom.z64, we should either use it as is or create a completely new rom that will be used as the base for extraction. Silently modifying it is very bad practice imo, and I would expect many users to be at least confused by it.

**Pull request closed without merging**